plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFieldsSchema.php

```diff
@@ -143,6 +153,189 @@ public function validate_schema( $options ) {
+									function ( $rate ) {
+										return $rate['rate_id'];
+									},
+									$package['shipping_rates']
+								);
+							},
+							$cart_data['shipping_rates']
+						)
+					)
+				),
+				'selected_shipping_rates' => $selected_shipping_rates_ids,
+				'prefers_collection'      => count( array_intersect( $local_pickup_method_ids, $selected_shipping_rates_methods ) ) > 0,
+				'items'                   => array_merge(
+					...array_map(
+						function ( $item ) {
+							return array_fill( 0, $item['quantity'], $item['id'] );
+						},
+						$cart_data['items']
+					)
+				),
+				'items_type'              => array_values(
+					array_unique(
+						array_map(
+							function ( $item ) {
+								return $item['type'];
+							},
+							$cart_data['items']
+						)
+					)
+				),
+				'needs_shipping'          => $cart_data['needs_shipping'],
+				'totals'                  => $cart_data['totals']->total_price,
+				'extensions'              => $cart_data['extensions'],
+			];
+		}
+
+		if ( $order ) {
+			$order_data = StoreApi::container()->get( SchemaController::class )->get( OrderSchema::IDENTIFIER )->get_item_response( $order );
+
+			$data['checkout'] = [
+				'order_id'           => $order_data['order_id'],
+				'customer_note'      => $order_data['customer_note'],
+				'additional_fields'  => $order_data['additional_fields'],
+				'payment_method'     => $order_data['payment_method'],
+				'needs_payment'      => $order->needs_payment(),
+				'available_gateways' => array_values( wp_list_pluck( WC()->payment_gateways->get_available_payment_gateways(), 'id' ) ),
+			];
+		}
+
+		if ( ! $customer ) {
+			$customer = $cart ? wc()->customer : new WC_Customer( $order->get_customer_id() );
+		}
+
+		if ( $customer || $billing_address || $shipping_address ) {
+			$billing_address  = $billing_address ? $billing_address : StoreApi::container()->get( SchemaController::class )->get( BillingAddressSchema::IDENTIFIER )->get_item_response( $customer );
+			$shipping_address = $shipping_address ? $shipping_address : StoreApi::container()->get( SchemaController::class )->get( ShippingAddressSchema::IDENTIFIER )->get_item_response( $customer );
+			$data['customer'] = [
+				'id'               => $customer->get_id(),
+				'guest'            => $customer->get_id() === 0,
+				'billing_address'  => $billing_address,
+				'shipping_address' => $shipping_address,
+				'address'          => 'billing' === $address_type ? $billing_address : $shipping_address,
+			];
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Check if the fields have a valid schema.
+	 *
+	 * @param array $fields The fields.
+	 * @return bool
+	 */
+	public function has_valid_schema( $fields ) {
+		$has_valid_schema = false;
+
+		foreach ( $fields as $field ) {
+			if (
+				isset( $field['rules'] ) &&
+				(
+					! empty( $field['rules']['required'] ) ||
+					! empty( $field['rules']['hidden'] ) ||
+					! empty( $field['rules']['validation'] )
+				)
```

Comment:
If a field in the array has invalid schema, but another has valid schema, this passes. Is this intentional?

Reply:
At this point I assume schema validation kicked in earlier and rejected this from running